extensions-contrib/druid-jwt-extensions/src/main/java/org/apache/druid/security/jwt/JwtJwksAuthenticator.java

```diff
@@ -0,0 +1,214 @@
+  public String getAuthChallengeHeader()
+  {
+    return "Bearer";
+  }
+
+  @Override
+  public Map<String, String> getInitParameters()
+  {
+    return null;
+  }
+
+  @Override
+  public String getPath()
+  {
+    return "/*";
+  }
+
+  @Override
+  public EnumSet<DispatcherType> getDispatcherType()
+  {
+    return null;
+  }
+
+
+  @Override
+  @Nullable
+  public AuthenticationResult authenticateJDBCContext(Map<String, Object> context)
+  {
+    String bearerToken = (String) context.get("bearer-token");
+
+    if (bearerToken != null) {
+      try {
+        JwtClaims jwtClaims = jwtVerifier.authenticate(bearerToken);
+        if (jwtClaims != null) {
+          LOG.info("JWT validation succeeded!");
+          return new AuthenticationResult(
+              JwtVerifier.getCallerGroup(jwtClaims),
+              authorizerName,
+              name,
+              null
+          );
+        }
+      }
+      catch (InvalidJwtException e) {
+        LOG.error(e, "Invalid Credentails");
+      }
+      catch (MalformedClaimException e) {
+        LOG.error(e, "Missing JWT Custom Claim: groups");
+      }
+    }
+    return null;
+  }
+
+  public class HTTPAuthenticationFilter implements Filter
+  {
+    @Override
+    public void init(FilterConfig filterConfig)
+    {
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+        throws IOException, ServletException
+    {
+      HttpServletResponse httpResp = (HttpServletResponse) servletResponse;
+      String bearerToken = AuthUtils.getBearerTokenFromHttpReq((HttpServletRequest) servletRequest);
+      if (bearerToken == null) {
+        // Request didn't have HTTP bearer token credentials, move on to the next filter
+        filterChain.doFilter(servletRequest, servletResponse);
+        return;
+      }
+
+      // filter JWT verifier based on issuer
+      // if issuer specified doesn't match to issuer from JWT then move on to the next filter
+      try {
+        String extractedIssuer = issuerExtractor.getIssuer(bearerToken);
+        if (!Objects.equals(extractedIssuer, issuer)) {
+          filterChain.doFilter(servletRequest, servletResponse);
+          return;
+        }
+      }
+      catch (InvalidJwtException | MalformedClaimException e) {
+        LOG.error(e, "Invalid JWT Issuer");
+        httpResp.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Invalid JWT Issuer");
+        return;
+      }
+
+      // If any jwt authentication error occurs, we send a 401 response immediately
+      // and do not proceed further down the filter chain.
+      try {
+        JwtClaims jwtClaims = jwtVerifier.authenticate(bearerToken);
+        LOG.info("JWT validation succeeded!");
+        // TODO add useful JWT claims into the context filed of AuthenticationResult
+        AuthenticationResult authenticationResult = new AuthenticationResult(
+            JwtVerifier.getCallerGroup(jwtClaims),
```

Comment:
Intent to change.

I want to map JWT Claims back to basic authorizer
`/druid-ext/basic-security/authorization/db/{authorizerName}/users/{userName}` so that we can reuse basic authorizer and keep it as is.
Mapping is redefined through configuration, like
```
druid.auth.authenticator.jwt.basicAuthZRoleMapping={"JWT claim":"authZ_username"}
```